ssv/committee_runner.go

```diff
@@ -0,0 +1,482 @@
+			}
+			partialMsg, err := cr.BaseRunner.signBeaconObject(cr, duty, attestationData, duty.DutySlot(),
+				types.DomainAttester)
+			if err != nil {
+				return errors.Wrap(err, "failed signing attestation data")
+			}
+			postConsensusMsg.Messages = append(postConsensusMsg.Messages, partialMsg)
+
+		case types.BNRoleSyncCommittee:
+			blockRoot := beaconVote.BlockRoot
+			partialMsg, err := cr.BaseRunner.signBeaconObject(cr, duty, types.SSZBytes(blockRoot[:]), duty.DutySlot(),
+				types.DomainSyncCommittee)
+			if err != nil {
+				return errors.Wrap(err, "failed signing sync committee message")
+			}
+			postConsensusMsg.Messages = append(postConsensusMsg.Messages, partialMsg)
+		}
+	}
+
+	ssvMsg := &types.SSVMessage{
+		MsgType: types.SSVPartialSignatureMsgType,
+		MsgID:   types.MessageID(cr.GetBaseRunner().QBFTController.Identifier),
+	}
+	ssvMsg.Data, err = postConsensusMsg.Encode()
+	if err != nil {
+		return errors.Wrap(err, "failed to encode post consensus signature msg")
+	}
+
+	msgToBroadcast, err := types.SSVMessageToSignedSSVMessage(ssvMsg, cr.BaseRunner.QBFTController.CommitteeMember.OperatorID,
+		cr.operatorSigner.SignSSVMessage)
+	if err != nil {
+		return errors.Wrap(err, "could not create SignedSSVMessage from SSVMessage")
+	}
+
+	if err := cr.GetNetwork().Broadcast(ssvMsg.MsgID, msgToBroadcast); err != nil {
+		return errors.Wrap(err, "can't broadcast partial post consensus sig")
+	}
+	return nil
+
+}
+
+func (cr CommitteeRunner) ProcessPostConsensus(signedMsg *types.PartialSignatureMessages) error {
+	// Gets all the roots that received a quorum of signatures
+	quorum, rootsList, err := cr.BaseRunner.basePostConsensusMsgProcessing(&cr, signedMsg)
+	if err != nil {
+		return errors.Wrap(err, "failed processing post consensus message")
+	}
+
+	if !quorum {
+		return nil
+	}
+
+	// Get unique roots to avoid repetition
+	rootSet := make(map[[32]byte]struct{})
+	for _, root := range rootsList {
+		rootSet[root] = struct{}{}
+	}
+
+	// Get validator-root maps for attestations and sync committees, and the root-beacon object map
+	attestationMap, committeeMap, beaconObjects, err := cr.expectedPostConsensusRootsAndBeaconObjects()
+	if err != nil {
+		return errors.Wrap(err, "could not get expected post consensus roots and beacon objects")
+	}
+
+	var anyErr error
+	attestationsToSubmit := make(map[phase0.ValidatorIndex]*phase0.Attestation)
+	syncCommitteeMessagesToSubmit := make(map[phase0.ValidatorIndex]*altair.SyncCommitteeMessage)
+
+	// For each root that got at least one quorum, find the duties associated to it and try to submit
+	for root := range rootSet {
+
+		// Get validators related to the given root
+		role, validators, found := findValidators(root, attestationMap, committeeMap)
+
+		if !found {
+			// Check if duty has terminated (runner has submitted for all duties)
+			if cr.HasSubmittedAllValidatorDuties(attestationMap, committeeMap) {
+				cr.BaseRunner.State.Finished = true
```

Comment:
maybe its best to `break` the loop here?

Reply:
Notice this comment on 203:
> All roots have quorum, so if we can't find validators for a root, it means we have a bug. We assume it is safe to stop due to honest majority assumption

But we've thought of it and breaking or even continuing is a better choice here so we can process other roots.
I'm adding this and a comment to explain this edge case.